Code/GraphMol/Chirality.cpp

```diff
@@ -2281,7 +2294,17 @@ void setBondStereoFromDirections(ROMol &mol) {
 
         bond->setStereoAtoms(startStereoAtom, endStereoAtom);
 
-        if (startDirBond->getBondDir() == endDirBond->getBondDir()) {
+        auto startBondDir = startDirBond->getBondDir();
```

Comment:
it looks weird to swap if the begin matches begin...

Maybe rename `startDirBond` to `beforeStartBond`? or `previousBond`?

    auto prevBondDir = previousBond->getBondDir();
    if (prevBondDir->getEndAtom() != startAtom) {
        prevBondDir = getOppositeBondDir(prevBondDir);
    }
    auto afterBondDir = afterBond->getBondDir();
    if (afterBond->getStartAtom() != endAtom) {
          afterBondDir = getOppositeBondDir(afterBondDir);
    }

Reply:
Well, there's some reason to it: every bond has a "begin" atom and an "end" atom. `startDirBond` is the directed bond at the "begin" side of the stereo bond. And we check if it is reversed if the begin atom of that bond is the same as the begin atom of the double bond...

Dropping the "Dir" part on the bond name is not a good idea, since there may be another bond at the "begin" side that does not have a direction set.

I'll update with some new names, hopefully more clarifying ones.